enhancements/installer/openstack-supported-versions.md

```diff
@@ -0,0 +1,226 @@
+  and possibly block the upgrade.
+
+### Design Details
+
+The OpenStack Compute service (also known as Nova) is always deployed
+in OpenStack clouds. Also, it is one of the services that bumps the
+microversions so regularly that is it possible to figure out
+what version of OpenStack is running based on the maximum version of
+Nova API that is available for a cloud.
+There's a contractual given here, which is: Nova will not bump the microversion
+of a stable release. There's also an assumption that Nova will always bump the
+microversion between stable releases (i.e. add or change the public API).
+This is a reasonable assumption which is unlikely to ever be broken.
+If it were ever not true we would still have an opportunity to look for
+alternatives.
+
+Therefore, it is proposed that we use that information as a reliable
+way of checking if the version of OpenStack that is running is
+actually supported.
+The version check should live in the Installer pre-flight checks near
+the other verifications that we do.
+
+The Nova API versions that are supported will be maintained in
+the installer code and will follow the recommendations from
+the [upstream Nova API manual](https://docs.openstack.org/nova/latest/reference/api-microversion-history.html).
+
+There are multiple reasons where an administrator would want to ignore
+the OpenStack version check:
+- the administrators know what they are doing and understand that at this
+  point they aren't supported anymore.
+- the administrators have performed a major upgrade or Fast-Forward-Upgrade
+  of OpenStack and the operation is partially finished; however
+  they need to proceed to an OpenShift operation (e.g. fresh install, scale up).
+  It it possible to run an OpenStack cloud with various versions of
+  APIs, as long as compatibility constraints are respected (out of
+  scope for this document).
+- the administrators have a Support Exception and an OpenStack upgrade
+  isn't possible at the moment of the operation.
+
+In that case, we would use a new configuration flag named `ignoreVersionCheck`.
+
+### User Stories
+
+#### Check the OpenStack version
+
+As an enterprise OpenShift cluster administrator, when I run an operation
+against my OpenShift cluster, I want to make sure that the version of
+OpenStack that is running, is actually supported.
+If not, I would like to see the operation to fail early.
+
+#### Ignore the version check
+
+As an enterprise OpenShift cluster administrator, I know what I'm doing
+and need to ignore the version check in a case where I absolutely need
+to install or upgrade OpenShift.
+Performing a scale operation of an unsupported OpenStack version would
+require one of:
+
+1. The underlying OpenStack has been downgraded
+2. OCP has removed support for the underlying OpenStack
+3. The underlying OpenStack has been upgraded to an unsupported version
+
+We don't support (1), (2) isn't likely to happen and (3) would imply
+an OpenStack regression.
```

Comment:
Not necessarily. 
Customer Context: Most customers have dedicated admin teams for products (IE: one for OSP one for OCP), and in large organizations, they may not know what the other is doing. So its feasible for the OSP team to upgrade the environment, and thus the OCP deployment is now running on an OSP deployment that is outside of the support matrix we support/certify for that version of OCP.